diff --git a/Assets/Prefabs/ItemDatabase.prefab b/Assets/Prefabs/ItemDatabase.prefab
index 7d640e7..cff63df 100644
--- a/Assets/Prefabs/ItemDatabase.prefab
+++ b/Assets/Prefabs/ItemDatabase.prefab
@@ -52,9 +52,11 @@ MonoBehaviour:
   m_Name: 
   m_EditorClassIdentifier: 
   Items:
-  - Value: 0
+  - Name: NULL
+    Value: 0
     Sprite: {fileID: 21300000, guid: 66770a3ad1f0a964f980449569e62dc2, type: 3}
     Category: -1
-  - Value: 0
+  - Name: DEBUG_100
+    Value: 100
     Sprite: {fileID: 21300000, guid: 66770a3ad1f0a964f980449569e62dc2, type: 3}
     Category: 0
diff --git a/Assets/Prefabs/Player.prefab b/Assets/Prefabs/Player.prefab
index d8f75a3..110b4b5 100644
--- a/Assets/Prefabs/Player.prefab
+++ b/Assets/Prefabs/Player.prefab
@@ -23,7 +23,7 @@ GameObject:
   - component: {fileID: 114457210784118820}
   - component: {fileID: 50827665106924822}
   - component: {fileID: 58606110347995994}
-  m_Layer: 0
+  m_Layer: 8
   m_Name: Player
   m_TagString: Untagged
   m_Icon: {fileID: 0}
@@ -39,7 +39,6 @@ GameObject:
   m_Component:
   - component: {fileID: 4208196472303254}
   - component: {fileID: 20566887986908272}
-  - component: {fileID: 212909814546087258}
   - component: {fileID: 81459772425091314}
   m_Layer: 0
   m_Name: Main Camera
@@ -172,6 +171,7 @@ MonoBehaviour:
   SprintMax: 1000
   CooldownSecs: 2
   NextSprint: 0
+  Score: 0
 --- !u!212 &212193363478648064
 SpriteRenderer:
   m_ObjectHideFlags: 1
@@ -218,49 +218,3 @@ SpriteRenderer:
   m_WasSpriteAssigned: 1
   m_MaskInteraction: 0
   m_SpriteSortPoint: 0
---- !u!212 &212909814546087258
-SpriteRenderer:
-  m_ObjectHideFlags: 1
-  m_CorrespondingSourceObject: {fileID: 0}
-  m_PrefabInternal: {fileID: 100100000}
-  m_GameObject: {fileID: 1899376032151500}
-  m_Enabled: 1
-  m_CastShadows: 0
-  m_ReceiveShadows: 0
-  m_DynamicOccludee: 1
-  m_MotionVectors: 1
-  m_LightProbeUsage: 1
-  m_ReflectionProbeUsage: 1
-  m_RenderingLayerMask: 4294967295
-  m_Materials:
-  - {fileID: 15303, guid: 0000000000000000f000000000000000, type: 0}
-  m_StaticBatchInfo:
-    firstSubMesh: 0
-    subMeshCount: 0
-  m_StaticBatchRoot: {fileID: 0}
-  m_ProbeAnchor: {fileID: 0}
-  m_LightProbeVolumeOverride: {fileID: 0}
-  m_ScaleInLightmap: 1
-  m_PreserveUVs: 0
-  m_IgnoreNormalsForChartDetection: 0
-  m_ImportantGI: 0
-  m_StitchLightmapSeams: 0
-  m_SelectedEditorRenderState: 0
-  m_MinimumChartSize: 4
-  m_AutoUVMaxDistance: 0.5
-  m_AutoUVMaxAngle: 89
-  m_LightmapParameters: {fileID: 0}
-  m_SortingLayerID: 0
-  m_SortingLayer: 0
-  m_SortingOrder: 0
-  m_Sprite: {fileID: 10907, guid: 0000000000000000f000000000000000, type: 0}
-  m_Color: {r: 1, g: 1, b: 1, a: 1}
-  m_FlipX: 0
-  m_FlipY: 0
-  m_DrawMode: 0
-  m_Size: {x: 0.16, y: 0.16}
-  m_AdaptiveModeThreshold: 0.5
-  m_SpriteTileMode: 0
-  m_WasSpriteAssigned: 1
-  m_MaskInteraction: 0
-  m_SpriteSortPoint: 0
diff --git a/Assets/Prefabs/WorldItem.prefab b/Assets/Prefabs/WorldItem.prefab
index 9dccb54..4991c0b 100644
--- a/Assets/Prefabs/WorldItem.prefab
+++ b/Assets/Prefabs/WorldItem.prefab
@@ -22,7 +22,7 @@ GameObject:
   - component: {fileID: 212744924815945730}
   - component: {fileID: 61973498492338110}
   - component: {fileID: 114213474810932792}
-  m_Layer: 0
+  m_Layer: 9
   m_Name: WorldItem
   m_TagString: Untagged
   m_Icon: {fileID: 0}
@@ -51,7 +51,7 @@ BoxCollider2D:
   m_Enabled: 1
   m_Density: 1
   m_Material: {fileID: 0}
-  m_IsTrigger: 0
+  m_IsTrigger: 1
   m_UsedByEffector: 0
   m_UsedByComposite: 0
   m_Offset: {x: 0, y: 0}
@@ -78,6 +78,7 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: c2410c34ed1237c4d842e1e6612fc37d, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
+  Value: 0
   Category: -1
   Renderer: {fileID: 0}
 --- !u!212 &212744924815945730
diff --git a/Assets/Scripts/ItemDatabase.cs b/Assets/Scripts/ItemDatabase.cs
index 41e23dd..f792f85 100644
--- a/Assets/Scripts/ItemDatabase.cs
+++ b/Assets/Scripts/ItemDatabase.cs
@@ -9,6 +9,7 @@ namespace Speedmarket
     [Serializable]
     public class Item
     {
+        public string Name;
         public int Value;
         public Sprite Sprite;
         public Category Category;
@@ -45,6 +46,7 @@ namespace Speedmarket
                 foreach(var worldItem in category)
                 {
                     int i = _rand.Next(categoryItems.Length);
+                    worldItem.Value = categoryItems[i].Value;
                     worldItem.Renderer.sprite = categoryItems[i].Sprite;
                 }
             }
diff --git a/Assets/Scripts/PlayerEntity.cs b/Assets/Scripts/PlayerEntity.cs
index e51f56c..809ec51 100644
--- a/Assets/Scripts/PlayerEntity.cs
+++ b/Assets/Scripts/PlayerEntity.cs
@@ -12,6 +12,13 @@ namespace Speedmarket
         [HideInInspector] public float CooldownSecs = 2.0f;
         [HideInInspector] public float NextSprint = 0.0f;
 
+        int _score;
+
+        public void UpdateScore(int val)
+        {
+            _score += val;
+        }
+
         void Update()
         {
             _controller.OldInput = _controller.NewInput;
diff --git a/Assets/Scripts/WorldItem.cs b/Assets/Scripts/WorldItem.cs
index 5524cf8..ad1cbe8 100644
--- a/Assets/Scripts/WorldItem.cs
+++ b/Assets/Scripts/WorldItem.cs
@@ -7,9 +7,12 @@ namespace Speedmarket
 {
     public class WorldItem : MonoBehaviour, IComparable<WorldItem>
     {
+        [HideInInspector] public int Value;
         public Category Category;
         [HideInInspector] public SpriteRenderer Renderer;
 
+        bool _triggered;
+
         public int CompareTo(WorldItem other)
         {
             int result = 0;
@@ -26,6 +29,18 @@ namespace Speedmarket
             return result;
         }
 
+        void OnTriggerEnter2D(Collider2D other)
+        {
+            if(_triggered)
+                return;
+
+            other.gameObject.GetComponent<PlayerEntity>()
+                .UpdateScore(Value);
+
+            Destroy(gameObject);
+            _triggered = true;
+        }
+
         void Start()
         {
             Renderer = GetComponent<SpriteRenderer>();
